backend/substrapp/tasks/tasks.py

```diff
@@ -1037,5 +1037,5 @@ def on_compute_plan(compute_plan):
         remove_algo_images(algo_hashes)
 
     # Remove intermediary models
-    if model_hashes:
+    if model_hashes and len(model_hashes) > 0:
```

Comment:
It's redundant to do so, if model_hashes = [], if model_hases is false in python

Reply:
Thanks! 

Then I don't understand how the hell we get log lines such as

```
[2020-07-28 19:29:09,637: INFO/MainProcess] Received task: substrapp.tasks.tasks.on_compute_plan[94da21f402df99382e1ae7e426654c287bba86b80b8d4e6d52c62b531944c5ef]
INFO 2020-07-28 19:29:09,655 substrapp.tasks.tasks 9 139843827689280 Remove intermediary models :
	-
[2020-07-28 19:29:09,657: INFO/ForkPoolWorker-1] Task substrapp.tasks.tasks.on_compute_plan[94da21f402df99382e1ae7e426654c287bba86b80b8d4e6d52c62b531944c5ef] succeeded in 0.019005998998181894s: None
```

Reply:
You can see in the following function : https://github.com/SubstraFoundation/substra-backend/blob/bcf2bbeae830f221f4a16e2bb40ccfeaeb47d7b3/backend/substrapp/tasks/tasks.py#L1014-L1023

You can see that `model_hashes` is not empty but the line is printed as your log because  `filtered_model_hashes` is empty. That means no model where find in this removal (there are not on this node but the other)

So you can change 
``` python
    log_model_hashes = '\n\t- '.join(filtered_model_hashes)
    logger.info(f'Remove intermediary models : \n\t- {log_model_hashes}')
```
to
``` python
    if filtered_model_hashes:
        log_model_hashes = '\n\t- '.join(filtered_model_hashes)
        logger.info(f'Remove intermediary models : \n\t- {log_model_hashes}')
```